python/ray/data/_internal/execution/operators/hash_shuffle.py

```diff
@@ -451,90 +476,129 @@ def start(self, options: ExecutionOptions) -> None:
         self._aggregator_pool.start()
 
     def _add_input_inner(self, input_bundle: RefBundle, input_index: int) -> None:
+
+        shuffle_metrics = self.get_metrics(0)
```

Comment:
Following up with my comment above -- let's re-home these multi-metrics set up in this class (being `_shuffle_stage_metrics` and `_reduce_stage_metrics`)